client.go

```diff
@@ -177,6 +182,27 @@ func (c *Client) Get(url string, resp interface{}) error {
 	return c.Do(req, nil, resp)
 }
 
+// Post is a convenience method that uses c.Do to issue a POST request to
+// the given URL. If c.BaseURL is configured then url will be treated as
+// being relative to c.BaseURL therefore should not contain a host part.
+// If params is non-nil then it will
+func (c *Client) Post(url string, params, resp interface{}) error {
```

Comment:
Given that most of this package is about marshaling HTTP request parameters,
I don't see why we're ignoring that here and just marshaling a JSON body (which is not even that conventional for POST requests).

Rather than this, I wonder about adding a CallURL method which would be exactly the same as Call except that it would take a URL parameter too, which would be treated the same as BaseURL for the duration of the request.

I *think* that would fit our current use case better than this.